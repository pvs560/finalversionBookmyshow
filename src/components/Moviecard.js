import React from "react";
import { CardDeck,Card,Button  } from 'react-bootstrap';
import PropTypes from "prop-types";

export default class MovieCard extends React.Component {

  constructor(props){
    super(props);
    console.log("moviecard constructor::",this.props.movielist);
  }


  onClick = (event)=>{
    console.log("book nlow clicked::",event.target.id);
    this.props.handleBookNow(event.target.id);
  }


    parseMovieList = (movielist)=>{
     return movielist.map(data =>         
     <Card key={data.imdbID} bsPrefix="card cardheight">
      <Card.Img variant="top" src={data.Poster} bsPrefix="card-img-top card-img-height"/>
      <Card.Body bsPrefix="card-body card-body-custom">
      <Card.Title>{data.Title}</Card.Title>
        <Card.Text>
          {data.Actors}
        </Card.Text>
      </Card.Body>
      <Card.Footer>
      <Button variant="primary" size="lg" block id={data.imdbID} onClick={this.onClick}>
Book Now
</Button>
      </Card.Footer>
    </Card>)
       
     }

    render(){
      console.log("moviecard in props::",this.props.movielist); 
     let parsedlist = this.parseMovieList(this.props.movielist);
      return ( <>
        <CardDeck bsPrefix="card-deck custom-card-deck">
        {parsedlist}
      </CardDeck>
      </>)
    }

}

MovieCard.protoTypes = {
  movielist:PropTypes.array.isRequired,
  handleBookNow:PropTypes.func
}