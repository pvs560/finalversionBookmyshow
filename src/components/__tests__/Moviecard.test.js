import React from 'react';
import renderer from 'react-test-renderer';

import Moviecard from '../Moviecard';
import MovieList from '../../JsonData/movielist';

describe('MovieComponent',()=>{
    test('sanpshot renders', ()=>{
        const component = renderer.create(<Moviecard movielist={MovieList.movielist} handleBookNow={()=>{}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});