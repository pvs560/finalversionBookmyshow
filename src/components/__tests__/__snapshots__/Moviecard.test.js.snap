// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieComponent sanpshot renders 1`] = `
<div
  className="card-deck"
>
  <div
    className="card"
  >
    <img
      className="card-img-top"
      src="https://m.media-amazon.com/images/M/MV5BMTMzMjMwMjcyNl5BMl5BanBnXkFtZTcwNTA1NDgzMQ@@._V1_SX300.jpg"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Avenger
      </div>
      <p
        className="card-text"
      >
        Sam Elliott, Timothy Hutton, James Cromwell, William Hope
      </p>
    </div>
    <div
      className="card-footer"
    >
      <button
        className="btn btn-primary btn-block btn-lg"
        disabled={false}
        id="tt0473445"
        onClick={[Function]}
        type="button"
      >
        Book Now
      </button>
    </div>
  </div>
  <div
    className="card"
  >
    <img
      className="card-img-top"
      src="https://m.media-amazon.com/images/M/MV5BMTQ3MDM1MDU2NF5BMl5BanBnXkFtZTgwMzM3OTIzNzE@._V1_SX300.jpg"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Race
      </div>
      <p
        className="card-text"
      >
        Stephan James, Jason Sudeikis, Eli Goree, Shanice Banton
      </p>
    </div>
    <div
      className="card-footer"
    >
      <button
        className="btn btn-primary btn-block btn-lg"
        disabled={false}
        id="tt3499096"
        onClick={[Function]}
        type="button"
      >
        Book Now
      </button>
    </div>
  </div>
  <div
    className="card"
  >
    <img
      className="card-img-top"
      src="https://m.media-amazon.com/images/M/MV5BZTkyYzc5MGEtYTBiYS00ZmYyLThlZWUtOWY3ZWE4ZDhlN2MzXkEyXkFqcGdeQXVyMjM0ODk5MDU@._V1_SX300.jpg"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Godfather
      </div>
      <p
        className="card-text"
      >
        N.N. Pillai, Mukesh, Kanaka, Philomina
      </p>
    </div>
    <div
      className="card-footer"
    >
      <button
        className="btn btn-primary btn-block btn-lg"
        disabled={false}
        id="tt0353496"
        onClick={[Function]}
        type="button"
      >
        Book Now
      </button>
    </div>
  </div>
</div>
`;
